php/php.blade/src/org/netbeans/modules/php/blade/editor/BladeDeclarationFinder.java

```diff
@@ -0,0 +1,463 @@
+                    location.addAlternative(new AlternativeLocationImpl(dlyieldItem));
+                }
+
+                return location;
+            case PUSH:
+            case PUSH_IF:
+            case PREPEND:
+                String stackId = referenceIdentifier;
+                List<BladeIndex.IndexedReference> stacks = QueryUtils.queryStacksReferences(stackId, currentFile);
+
+                if (stacks == null) {
+                    return DeclarationLocation.NONE;
+                }
+
+                for (BladeIndex.IndexedReference stackReference : stacks) {
+                    String stackReferenceId = stackReference.getReference().identifier;
+                    NamedElement yieldIdHandle = new NamedElement(stackReferenceId, stackReference.getOriginFile(), ElementType.STACK_ID);
+                    int startOccurence = stackReference.getReference().defOffset.getStart();
+                    DeclarationLocation dlstack = new DeclarationFinder.DeclarationLocation(stackReference.getOriginFile(), startOccurence, yieldIdHandle);
+                    if (location.equals(DeclarationLocation.NONE)) {
+                        location = dlstack;
+                    }
+                    location.addAlternative(new AlternativeLocationImpl(dlstack));
+                }
+
+                return location;
+            case CUSTOM_DIRECTIVE:
+                String directiveNameFound = reference.identifier;
+                DeclarationLocation dlcustomDirective = DeclarationLocation.NONE;
+
+                CustomDirectives.getInstance(projectOwner).filterAction(new CustomDirectives.FilterCallbackDeclaration(dlcustomDirective) {
+                    @Override
+                    public void filterDirectiveName(CustomDirective directive, FileObject file) {
+                        if (directive.name.equals(directiveNameFound)) {
+                            NamedElement customDirectiveHandle = new NamedElement(directiveNameFound, file, ElementType.CUSTOM_DIRECTIVE);
+                            DeclarationFinder.DeclarationLocation newLoc = new DeclarationFinder.DeclarationLocation(file, directive.offset, customDirectiveHandle);
+                            this.location.addAlternative(new AlternativeLocationImpl(newLoc));
+                        }
+                    }
+                });
+
+                if (!dlcustomDirective.getAlternativeLocations().isEmpty()) {
+                    for (AlternativeLocation loc : dlcustomDirective.getAlternativeLocations()) {
+                        dlcustomDirective = loc.getLocation();
+                    }
+                }
+                return dlcustomDirective;
+            case PHP_CLASS:
+                Collection<PhpIndexResult> indexClassResults;
+                String namespace = reference.namespace;
+
+                if (namespace != null && reference.namespace.length() > 2) {
+                    int subOffset = reference.namespace.startsWith("\\") ? 1 : 0;
```

Comment:
I would add `"\\"` to somewhere(e.g. Utils) as a constant.

e.g. `public static final String NAMESPASE_SEPARATOR = "\\"; // NOI18N`

Reply:
Don't we have already somewhere?